eeps/eep-0057.md

```diff
@@ -0,0 +1,353 @@
+
+## Compilation
+
+Compiling a module with the `-moduledoc` or `-doc` attributes will
+automatically generate a Docs chunk into its .beam file, making the
+documentation directly accessible in the shell.  A `-nodocs` flag can
+be added to `erlc` to skip the generation of the chunk.
+
+Release tools should also prune the Documentation chunk out of .beam
+files by default.  Note this is already done by `beam_lib:strip_release/1`
+and `beam_lib:strip_files/1`.
+
+Documentation format
+====================
+
+One big discussion about documentation is what is the documentation
+format that the documentation should adopt.  Luckily, EEP 48 is agnostic
+to the format, however one must still be listed.  There are a couple
+options to this problem:
+
+  1. The Erlang/OTP picks a documentation format as the default for
+    the Erlang community, such as AsciiDoc.
+
+  2. The documentation format must be explicitly listed via a new
+    attribute called `-docformat`, such as `-docformat "text/asciidoc".`
+
+This proposal does not attempt to discuss the prefered documentation
+format, as it is a separate discussion. It is also possible to support
+both options above (pick a format but allow it to be overridden).
+
+Other topics
+============
+
+This section is going to list other topics, and future extensions,
+that are related to this topic but are not required for the
+implementation of this EEP.
+
+## Unicode binary strings
+
+In the examples above, we have used char lists for the documentation.
+However, the Docs chunk requires the documentation to be a binary
+string. We have a couple of options:
+
+  * Support binaries exclusively
+  * Support lists exclusively with automatic conversion to binaries
+    at some point
+  * Support both prior formats and automatically convert lists to
+    binaries at some point - preferably in the parser/compiler,
+    to avoid pushing this concern to everyone downstream
+
+Note that similar concerns apply to documentation metadata.
+Attributes such as License, Authors, and so on, are required
+to be binaries.  Since those attributes are known upfront, they
+can be validated/normalized by the parser/compiler.
+
+With this in mind, it is important to remember that Erlang binaries
+are latin1 by default. Therefore, if the documentation contains
+non-latin1 characters, one would have to write:
+
+    -doc <<"Documentation about púnctuation"/utf8>>.
+
+Similarly, the Authors metadata:
+
+    -doc #{
+      authors => [<<"José Valim"/utf8>>]
+    }.
+
+For this reason, the Build and Packaging Working Group of the
+Erlang Ecosystem Foundation has also discussed the option of making
+Unicode binary strings easier to write in Erlang, such as
+`u"Héllò World"`.  While adding such construct is not a direct part
+of this EEP, the point made in this subsection is that, if we require
+the documentation or its metadata to be a binary, we may need to
+provide conveniences for writing UTF-8 binary strings in Erlang.
```

Comment:
This, to me, would point towards supporting both formats and converting. Erlang strings should represent safe unicode data that can be converted later with `unicode:characters_to_binary(CharData)` and for the time being provides safe unicode with low syntactic overhead.

Reply:
Yeah, I think I agree with you. What about metadata? Should it be automatically rewritten too?

Reply:
It could make sense. I'd see this as an extension of all Erlang-style things where `CharData` gets flattened down for you when possible. It would in principle not harm anything as long as you specify you expect data to be UTF-8 compatible, meaning raw strings, binaries in the utf8 encoding, or lists of either of these.